scipio/src/io/dma_file.rs

```diff
@@ -517,12 +518,21 @@ impl DmaFile {
     }
 
     /// Closes this DMA file.
-    pub async fn close(&mut self) -> io::Result<()> {
+    pub async fn close(mut self) -> io::Result<()> {
         let source = Reactor::get().close(self.as_raw_fd());
         enhanced_try!(source.collect_rw().await, "Closing", self)?;
         self.file = unsafe { std::fs::File::from_raw_fd(-1) };
         Ok(())
     }
+    pub(crate) async fn close_rc(self: Rc<DmaFile>) -> io::Result<()> {
```

Comment:
Note that  `Rc<Self>` works with method call syntax.

Reply:
this is indeed better.

All good and no objections. Pushing the button on this, thanks!